### DISCUSSION TITLE
supabase edge function in express.js

### DISCUSSION THREAD

**Discussion by damiankomonski** [supabase/supabase#21821] [Questions]
Hello, I have question. Is it possible to write edge function in Express.js? Not using Deno?

I saw some example in documentation but I am thinking if it is possible.
https://supabase.com/docs/guides/functions/routing#routing-with-express
 
 Please help, I remember that in Firebase I could use express.js, write application and pass this app.

**Comment by damiankomonski** [comment]
In one example I am seeing app.listen, in another it isnt there. 

One question yet:
`Deno.env.get('SUPABASE_URL')`
How can I use environment variables in functions when I am using express.js?

**Reply by NemoTalenthub** [reply]
You need to add app.listen anywhere you use, and this is how you use env, Just add these env to supabase secret on edge function dashboard

`import express, { Request, Response } from "npm:express@4.18.2";
import fetch from "npm:node-fetch@3.3.2";
import process from "node:process";

const app = express();

// Ensure environment variables are loaded
const baseURL = process.env.BASE_URL || "";
const userAgent = process.env.USER_AGENT || "";
const username = process.env.USERNAME || "";
const password = process.env.PASSWORD || "";`
